### DISCUSSION TITLE
How to model (nearly) full-truckload pickup and delivery?

### DISCUSSION THREAD

**Discussion by charlestang06** [PyVRP/PyVRP#605] [Q&A] [answered]
I'm wondering if the following scenario can be modeled with the PyVRP package and how--the most resembling VRPSPD has a central depot which I do not want to use.

Scenario:
- 20-30 vehicles each with capacity 2
- Total of 100-ish locations for both delivery and drop off
- a long task list of pick up and delivery (drop off) locations with quantities > 2 (meaning multiple vehicles may need to visit a location multiple times)
- Vehicle can continue its route after dropping off load at delivery location

I am wondering because (a) there is no central depot(s) and (b) pick up/delivery quantities are greater than capacity.

Thanks

**Comment by N-Wouda** [comment, marked as answer]
Hi @charlestang06! Sounds like an interesting setting. We do not support this in full generality, but if I understand you correctly you can get a fairly good approximation by modelling it as follows. 

Although you do not have a central depot, PyVRP requires at least one. Just add a dummy depot with default time windows, and set all edge distances and durations from and to that depot to 0.

I assume you have a list of $(O, D, q)_i$-triplets for each shipment $i$, where $O_i$ is an origin location, $D_i$ a destination, and $q_i > 2$ some demand. We are going to do a full-truckload approximation. Simply count $n_i = \lceil q_i / 2 \rceil$, the minimum number of shipments that are needed to move $q_i$ from $O_i$ to $D_i$. 

Add $n_i$ dummy clients with zero demand to your model, each at the location $D_i$. These replace the $O_i$ and $D_i$ locations for shipment $i$. Arcs leaving any of the new clients are the same as those originally leaving $D_i$, but any arcs _into_ the new clients need to have a distance and duration that takes into account that on the arc we first (implicitly) go to $O_i$ to pick up a full truckload of demand. The picked up demand is then immediately dropped of at $D_i$ through one of the dummy clients. This should hopefully get you a good approximate solution.
